File: THCS/2022_2023/DakLak_2223/Bai2/BAI2.cpp
#include <iostream>
#include <vector>

using namespace std;

// Hàm kiểm tra số nguyên n có phải là số nguyên tố hay không
bool is_prime(int n) {
    if (n <= 1) return false;
    if (n <= 3) return true;
    if (n % 2 == 0 || n % 3 == 0) return false;
    for (int i = 5; i * i <= n; i += 6) {
        if (n % i == 0 || n % (i + 2) == 0) return false;
    }
    return true;
}

// Hàm kiểm tra số nguyên n có phải là số nguyên tố "đẹp" hay không
bool is_beautiful_prime(int n) {
    if (!is_prime(n)) return false;
    // Kiểm tra xem n có chứa chữ số 3, 5, hoặc 7 không
    while (n > 0) {
        int digit = n % 10;
        if (digit == 3 || digit == 5 || digit == 7) return false;
        n /= 10;
    }
    return true;
}

// Hàm đếm số lượng số nguyên tố "đẹp" từ 1 đến N
int count_beautiful_primes(int N) {
    int count = 0;
    for (int i = 1; i <= N; ++i) {
        if (is_beautiful_prime(i)) {
            count++;
        }
    }
    return count;
}

int main() {
    int N;
    cout << "Nhập số nguyên dương N: ";
    cin >> N;

    int result = count_beautiful_primes(N);
    cout << result << endl;

    return 0;
}

File: THCS/2022_2023/DakLak_2223/Bai2/BAI2.py
def is_prime(n):
    """Kiểm tra số nguyên n có phải là số nguyên tố hay không."""
    if n <= 1:
        return False
    if n <= 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False
    i = 5
    while i * i <= n:
        if n % i == 0 or n % (i + 2) == 0:
            return False
        i += 6
    return True

def is_beautiful_prime(n):
    """Kiểm tra số nguyên n có phải là số nguyên tố "đẹp" hay không."""
    if not is_prime(n):
        return False
    # Kiểm tra xem n có chứa chữ số 3, 5, hoặc 7 không
    str_n = str(n)
    return '3' not in str_n and '5' not in str_n and '7' not in str_n

def count_beautiful_primes(N):
    """Đếm số lượng số nguyên tố "đẹp" từ 1 đến N."""
    count = 0
    for i in range(1, N + 1):
        if is_beautiful_prime(i):
            count += 1
    return count

# Đọc số nguyên dương N từ bàn phím
N = int(input("Nhập số nguyên dương N: "))
result = count_beautiful_primes(N)

# Xuất kết quả
print(result)
